backend/src/users/users.controller.ts

```diff
@@ -10,10 +10,14 @@ export class UsersController {
     constructor(private readonly usersService: UsersService) {}
 
     @Post()
-    async create(@Body() createUserDto: CreateUserDto) {
-        const user = await this.usersService.create(createUserDto);
-        const url = createConfirmationUrl(user['_id'])
-        await sendEmail(user.email, url)
+    async create(@Body() userDto: CreateUserDto) {
+        if(userDto['login']) {
+            return await this.usersService.login(userDto['login']);
+        } else {
+            const user = await this.usersService.create(userDto);
```

Comment:
Are you signing up users when they are logging in if the account does not exist?

Reply:
When an account does not exist I throw the "Username / password combination is incorrect" message.

Reply:
What does `else` block do here?

Reply:
For creating a new user.

Reply:
Yes, creating a new user should be a separate method.